src/Microsoft.AspNetCore.Rewrite/Internal/ApacheModRewrite/ApacheModRewriteRule.cs

```diff
@@ -9,10 +9,10 @@ namespace Microsoft.AspNetCore.Rewrite.Internal.ApacheModRewrite
     public class ApacheModRewriteRule : IRule
     {
         public UrlMatch InitialMatch { get; }
-        public IList<Condition> Conditions { get; }
+        public ConditionCollection Conditions { get; }
```

Comment:
mod rewrite uses different conditional logic where the and/or is per condition. What you're doing to UrlRewrite does not apply. Revert.